### PR TITLE
feat!: Adding support for Moving Platorms

### PR DESCRIPTION
**Pull request by kurtome**
This new feature is implemented as a new object type users can add to the Tiled map directly, with positioning, path, and speed control in the Tiled object's custom properties. Implementers need to render the sprite themselves.

This PR required refactoring the collision detection to include non-tiles in the ground detection logic, and adding `tags` to the PhysicalEntity class so entities can specify which other objects they consider solid (like the ground).

### CODE REVIEW

File: packages/leap/lib/src/entities/moving_platform.dart [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+  /// Path of this platform, described as list of nodes of tile offsets.
+  ///
+  /// For example, the tuple {-2, 4} would indicate that the platform should
+  /// move 2 tiles to the left and 4 tiles down from its last node in the
+  /// tile path.
+  ///
+  /// The initial position of the platform is implicitly considered the current
+  /// node when the platform is created.
+  final List<Tuple2<int, int>> tilePath;
```
Comment (erickzanardo) [inline packages/leap/lib/src/entities/moving_platform.dart:53]: We can probably use records instead? Which is native to the dart language?

Also, regarding this same subject, I implemented an expression language that I am using to define paths in my games, this seems like a good place to use it: https://github.com/erickzanardo/pathxp

The advantages of that expression language over a simple tuple, is that the developer can define if the path should repeat, or if it is infinite, while also defining more complex paths. Let me know what you think

Comment (kurtome) [reply packages/leap/lib/src/entities/moving_platform.dart:53]: Does `pathxp` support diagonal movement? Like `2T2R` for two up and two right in one step?

It does seem like a nearly perfect fit for this usecase

Comment (erickzanardo) [reply packages/leap/lib/src/entities/moving_platform.dart:53]: It doesn't, but I am happy to implement it

**Comment by kurtome** [comment]
> Still have the open suggestion to use pathxp for the path of the platforms (and maybe even for enemies in the future?), but since pathxp still don't have a diagonal notation, maybe we want to merge as is and implement pathxp support later.


I was just thinking the same thing

**Pull request merged**